src/main/java/mainFiles/UserInput.java

```diff
@@ -0,0 +1,81 @@
+package mainFiles;
+
+/**
+ * Class for receiving user input and storing it as separate arguments.
+ */
+public class UserInput {
+    private final String name;
+    private final String arg1;
+    private final String arg2;
+    private final String arg3;
```

Comment:
do you think it would be clearer to name these instances fromDate, byDate, and toDate respectively?

Reply:
Ah, in this case I've left these as more generic arguments because they're meant to support other things, so for example, in the case of `mark`ing, it accepts the input as `mark x`, but x is an integer, not necessarily a date.